.github/workflows/canister-tests.yml

```diff
@@ -534,3 +534,18 @@ jobs:
           # populated by GitHub Actions
           # https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  clean-build:
+    runs-on: ${{ matrix.os }}
+    if: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
Is there a CI run that executed this action (e.g. before the condition was introduced) that you could link to?

Reply:
There you go: https://github.com/dfinity/internet-identity/actions/runs/2495144575

It's not exactly the same, but very similar; also note that the workflow got canceled after I force pushed the commit